src/database/DataTypes/Item.js

```diff
@@ -224,6 +216,96 @@ export class Item extends Realm.Object {
   toString() {
     return `${this.code} - ${this.name}`;
   }
+
+  /**
+   * Finds all this items batches in a specific location.
+   * If no location is passed, all of this items ItemBatches
+   * are returned.
+   * @param {Location} location
+   */
+  getBatchesInLocation({ id } = {}) {
+    if (!id) return this.batches;
+    return this.batches.filtered('location.id = $0', id);
+  }
+
+  /**
+   * Returns the number of batches in a specific location. If
+   * no location is passed, finds the number of batches for all
+   * batches.
+   * @param {Location} location
+   */
+  getNumberOfBatches(location) {
+    return this.getBatchesInLocation(location).length;
+  }
+
+  /**
+   * Returns the sum of each ItemBatch.totalQuantity in a specific
+   * location. If no location is passed, finds the totalQuantity
+   * for all batches.
+   * @param {Location} location
+   */
+  getQuantityInLocation(location) {
+    return this.getBatchesInLocation(location).reduce(
+      (sum, { numberOfPacks = 0, packSize = 1 } = {}) => sum + numberOfPacks * packSize,
+      0
+    );
+  }
+
+  /**
+   * Returns all ItemBatches which have been in a breach in a specific
+   * location. If no location is passed, all of this items ItemBatches
+   * in breaches are returned.
+   * @param {Location} location
+   */
+  getBreachedBatches({ id } = {}) {
+    const breachedBatches = this.batches.filter(({ hasBreached } = {}) => hasBreached);
```

Comment:
do't need { hasBreached } `= {}`, this.batches should only contain realm objects.

by the way, can you not use getBatchesInLocation(location).filter ...

Reply:
done 7a71e8d33c0263d93c30efd490959976131d834c